src/aiida_quantumespresso/workflows/pw/base.py

```diff
@@ -563,16 +564,36 @@ def handle_electronic_convergence_not_reached(self, calculation):
 
         Decrease the mixing beta and fully restart from the previous calculation.
         """
-        factor = self.defaults.delta_factor_mixing_beta
+        import numpy
+
+        scf_accuracy = calculation.tools.get_scf_accuracy(0)
+        scf_accuracy_slope = numpy.polyfit(numpy.arange(0, len(scf_accuracy)), numpy.log(scf_accuracy), 1)[0]
```

Comment:
I would also take the _intercept_ to use for extrapolating the amount of extra steps to perform. Or otherwise, we can simply increase it to a fixed maximum? E.g. `electrons_maxstep = self.defaults.max_electronic_steps`?